### ISSUE TITLE
IconStyle hotspot

### ISSUE THREAD

**Issue by ardouglas** [georust/kml#11]
Hi and thank you so much for the great library!  I've been using it quite a bit, but found that I needed support for the `hotSpot` element that can be present on the `IconStyle` tag.  I have a this working on a private fork and would like to contribute upstream if you are interested.  My current implementation took the naive approach of making a tuple to represent the element and updating the necessary reader/writer logic:

``` rust
#[derive(Clone, Debug, PartialEq)]
pub struct IconStyle {
    pub id: String,
    pub scale: f64,
    pub heading: f64,
    pub hot_spot: Option<(f64, f64, String, String)>,
    pub icon: Icon,
    pub color: String,
    pub color_mode: ColorMode,
}
```

While this does work, I suspect a better API would be to create the underlying structure and units enum for the `kml:vec2Type`.

```
<complexType name="vec2Type" abstract="false">
    <attribute name="x" type="double" default="1.0"/>
    <attribute name="y" type="double" default="1.0"/>
    <attribute name="xunits" type="kml:unitsEnumType" use="optional"
      default="fraction"/>
    <attribute name="yunits" type="kml:unitsEnumType" use="optional"
      default="fraction"/>
  </complexType>
```

If you're interested I'd be glad to take this on myself.  Thanks!

**Comment by pjsier** [comment]
Thanks for posting this! This definitely sounds interesting, so any contribution would be welcome.

I think you're probably right about creating a new struct for `vec2Type` so if you have the time that would be great as well

**Comment by ardouglas** [comment]
I have started an implementation here: [rough draft](https://github.com/ardouglas/kml/commit/9387b0855989bf786f83d901168e4c68e8aed8ea).  I wanted to check with you before proceeding to docs and tests to see if I am following the right path?  I'm thinking you might prefer different naming or package layout conventions from what I've chosen.  Thanks again!

**Comment by pjsier** [comment]
@ardouglas this looks great! Feel free to keep going and open a PR when you're ready

**Comment by pjsier** [comment]
Closed by #13, thanks again @ardouglas !
